managed/src/main/java/com/yugabyte/yw/models/helpers/CustomerConfigValidator.java

```diff
@@ -179,4 +194,48 @@ public ObjectNode validateDataContent(JsonNode formData) {
         formData.get("data"), errorJson));
     return errorJson;
   }
+
+  /**
+   * Validates data which is contained in formData.
+   * During the procedure it checks all the S3 config parameters whether they are valid or not.
+   * Errors are collected and
+   * returned back as a result. Empty result object means no errors.
+   *
+   * Currently is checked:
+   *  - S3/AWS - S3 Bucket, whether it exists or not.
+   *
+   * @param formData, region
+   * @return Json filled with errors
+   */
+  public static ObjectNode validateS3DataContent(JsonNode formData, ObjectNode errorJson) {
+    try {
+      String bucketname = formData.get("BACKUP_LOCATION").asText();
+      //Assuming bucket name will always start with s3:// otherwise that will be invalid
+      bucketname = bucketname.substring(5);
+      AmazonS3 s3client = getS3Client(formData);
+      List<Bucket> buckets = s3client.listBuckets();
+      for (Bucket bucket : buckets) {
+        if (bucket.getName().equals(bucketname))
+          return errorJson;
+      }
+      errorJson.set("BackupConfigException", Json.newArray().add("Bucket name " + bucketname +
+          " doesn't exist"));
+    } catch (AmazonS3Exception s3Exception) {
+      errorJson.set("BackupConfigException", Json.newArray().add(s3Exception.getErrorMessage()));
+    }
+    return errorJson;
+  }
+
+  // TODO: move to some common utils or AWSCloudImpl file
+  public static AmazonS3 getS3Client(JsonNode formData) {
```

Comment:
create 
interface S3ClientFactory {
   AmazonS3 create(String key, String secret);
}

Then implement that factory (code from this method)
Then  take S3ClientFactory in ctor (DI)

In tests you can inject mocks3clientfactory that gives mock s3client 
you can then test based on various listBuckets behaviors failures as well as success cases.

The comment is so you know how to unit test this type of change. 
It would be nice if you add such test but its okay if you just do manual testing by bringing up the server.

Reply:
We can move this AmazonS3 create(String key, String secret); to CloudAPI.java interface.
And will provide there implementation in AwsCloudImpl.java.
Does this sound OK? 

I don't think we should create an interface for this since we already have an interface created.
Please suggest! Thanks!

https://github.com/yugabyte/yugabyte-db/blob/master/managed/src/main/java/com/yugabyte/yw/cloud/CloudAPI.java

Reply:
That seems reasonable.